activiti-core/activiti-image-generator/src/main/java/org/activiti/image/impl/DefaultProcessDiagramGenerator.java

```diff
@@ -193,16 +206,25 @@ public void draw(DefaultProcessDiagramCanvas processDiagramCanvas,
                 if (throwEvent.getEventDefinitions() != null && !throwEvent.getEventDefinitions().isEmpty()) {
                     if (throwEvent.getEventDefinitions().get(0) instanceof SignalEventDefinition) {
                         processDiagramCanvas.drawThrowingSignalEvent(flowNode.getId(),
+                                                                     flowNode.getName(),
                                                                      graphicInfo);
                     } else if (throwEvent.getEventDefinitions().get(0) instanceof CompensateEventDefinition) {
                         processDiagramCanvas.drawThrowingCompensateEvent(flowNode.getId(),
+                                                                         flowNode.getName(),
                                                                          graphicInfo);
-                    } else {
+                    } else if(throwEvent.getEventDefinitions().get(0) instanceof LinkEventDefinition) {
```

Comment:
```suggestion
                    } else if(throwEvent.isLinkThrowEvent()) {
```
BTW, I think you can evolve the implementation of `isLinkThrowEvent` and `isLinkCatchEvent` to simplify the logic:
- Create a method `isLinkEvent` directly inside the class `Event`:
- Update methods `isLinkThrowEvent` and `isLinkCatchEvent` to use `isLinkEvent` method once the cast to `IntermediateCatchEvent` or `ThrowEvent` is done.
- here you can use directly `isLinkEvent`